octoprint_PrintTimeGenius/static/js/PrintTimeGenius.js

```diff
@@ -70,19 +72,38 @@ $(function() {
         }
       }
       return results;
-    }
+    };
 
-    self.FileList = ko.pureComputed(function() {
-      return self.recurseFiles(self.filesViewModel.allItems()).slice()
-          .sort(function(a,b) {
-            if (_.has(a, "gcodeAnalysis.progress") !=
-                _.has(b, "gcodeAnalysis.progress")) {
-              return (_.has(a, "gcodeAnalysis.progress") -
-                      _.has(b, "gcodeAnalysis.progress"));
-            }
-            return a.path.localeCompare(b.path);
-          });
-    });
+    self.theFiles = function(items) {
+    	let results = [];
+    	let queue = [{children: items}];
+
+    	while (queue.length > 0) {
+    		item = queue.shift();
+    		results.push.apply(results, item.children.filter(item => item["type"] == "machinecode"));
+    		queue.push.apply(queue, item.children.filter(item => "children" in item));
+    	}
+    	return results;
+    };
+
+    self.update_filelist = function() {
+        self.FileList(self.theFiles(self.filesViewModel.allItems()));
+    };
+
+    self.onEventUpdatedFiles = function(payload) {
+        if (self.settings_visible() && payload.type === 'printables') {
+            setTimeout(self.update_filelist, 20000);
```

Comment:
this has a 20 second wait before updating because per my testing, the `UpdateFiles` event always fires *before* the file list is actually updated. Sometimes it's only 2 seconds before, and sometimes it's as much as 15 seconds before. So, waiting 20 seconds seems to account for it

this could be considered an OctoPrint bug though, I'm not really sure what the expectation is supposed to be

Reply:
I guess we could ask.  @foosel, have you heard of this?  I didn't see any bug reports.